apis/datasciencecluster/v1/datasciencecluster_types.go

```diff
@@ -98,7 +135,7 @@ type DataScienceClusterStatus struct {
 
 	// Expose component's specific status
 	// +optional
-	Components ComponentsStatus `json:"components,omitempty"`
+	Components ComponentsStatus `json:"components"`
```

Comment:
Minor nit - IMHO it would be clearer to name this 'ComponentsStatus', rather than 'Components' so one can infer what type the field is at a glance  from its name and easily distinguish it from `InstalledComponents` .  You can safely ignore me if this is convention you have agreed on and are following :)

Reply:
This is already part of the DSC API so it not possible to rename it unless we introduce a new API version :(